src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -322,6 +323,61 @@ export function VislibVisualizationsPieChartProvider(Private) {
       }
     }
 
+    readableNumber(number) {
+      let tier = Math.log10(Math.abs(number)) / 3 | 0;
+      if(tier === 0) return number;
+      if(tier > 4) tier = 4;
+      const postfix = this.ABBREVIATIONS[tier];
+      const scale = Math.pow(10, tier * 3);
+      const scaled = number / scale;
+      let formatted = String(scaled.toFixed(1));
+      if (/\.0$/.test(formatted)) {
+        formatted = formatted.substr(0, formatted.length - 2);
+      }
+      return formatted + postfix;
+    }
+
+    addDonutDescriptor(svg, width, height) {
+      const self = this;
+      const metricLabel = self.events.handler.data.data
+        .raw.columns.find(function (element) {
+          return element.categoryName === 'metric';
+        }).label;
+      const descriptor = svg.append('text')
+        .style('text-anchor', 'middle')
+        .attr('font-size', function (chartData) {
+          const dividend = Math.min(width, height);
+          let length = 0;
+          if(chartData.raw) {
+            length = chartData.raw.columns.length;
+          }
+          const divisor = Math.max(length, 4);
+          return (dividend / divisor) + 'px';
+        })
+        .attr('dy', '.35em')
+        .text(function (chartData) {
+          return self.readableNumber(chartData.slices.sumOfChildren);
+        });
+      let tooltip;
```

Comment:
In my opinion we could get rid of the tooltip altogether. The user already enabled "Show total", so I would assume they would be aware, that it's the total of the chart. If we want to keep the tooltip, we need to add a better styling for it, because currently it's having no padding at all. A solution could be adding a table into it with a "Total {label}" and a "{value}" column in it.

In general I don't like that we mess around with the tooltip like that. We don't try to manually modify the tooltip in any other chart like that, so I would really prefer getting rid of it for now.

Reply:
Generally, the person creating the dashboards isn't the one using them. It would aid dashboard users. Perhaps I should add a table? Or we have a better way to add such a tooltip?